src/Gql.js

```diff
@@ -0,0 +1,266 @@
+		}
+	}
+
+	return execute(
+		schema,
+		document,
+		rootValue,
+		context,
+		variables,
+	);
+}
+
+function convertObjToGqlArg(obj) {
+	const gqlArg = [];
+	_.forEach(obj, (value, key) => {
+		// eslint-disable-next-line no-use-before-define
+		gqlArg.push(`${key}: ${convertToGqlArg(value)}`);
+	});
+	return `${gqlArg.join(', ')}`;
+}
+
+function convertStrToGqlArg(str) {
+	const [, val] = str.match(ENUM_REGEX) || [];
+	if (val) return val;
+	// const [, prefix, val] = str.match(NO_QUOTES_REGEX) || [];
+	// if (prefix && val) return prefix === VAR_PREFIX ? '$' + val : val;
+
+	return JSON.stringify(str);
+}
+
+function convertToGqlArg(value) {
+	if (value == null) return null;
+
+	if (typeof value === 'number') return String(value);
+	if (typeof value === 'string') return convertStrToGqlArg(value);
+	if (_.isPlainObject(value)) return `{${convertObjToGqlArg(value)}}`;
+
+	return JSON.stringify(value);
+}
+
+class ApiError extends Error {}
+class GraphqlError extends Error {}
+
+class Gql {
+	constructor(opts = {}) {
+		if (opts.api) {
+			this.api = _.defaults(opts.api, {
+				endpoint: null,
+				headers: {},
+				cookies: {},
+			});
+		}
+		else {
+			const {schema, pubsub, defaultSchema} = makeSchemaFromConfig(opts);
+
+			this.schema = schema;
+			this.defaultSchema = defaultSchema;
+			this.pubsub = pubsub;
+			this.validateGraphql = opts.validateGraphql || false;
+			this.formatError = opts.formatError || formatError;
+		}
+
+		this.cache = opts.cache;
+	}
+
+	async _execApi(query) {
+		const response = await Connect
+			.url(this.api.endpoint)
+			.headers(this.api.headers)
+			.cookies(this.api.cookies)
+			.body({query})
+			.post();
+
+		const result = Str.tryParseJson(response.body);
+
+		if (response.statusCode !== 200) {
+			throw new ApiError(`${response.statusCode}, ${(result && result.errors) || 'Unknown error'}`);
+		}
+
+		if (!result) {
+			throw new ApiError('Invalid result from api');
+		}
+
+		if (!_.isEmpty(result.errors)) {
+			const err = new ApiError('Errors in api response');
+			err.errors = result.errors;
+			throw err;
+		}
+
+		return result;
+	}
+
+	async _execGraphql(query, context, {variables = {}, schemaName} = {}) {
+		const schema = schemaName ? this.schema[schemaName] : this.defaultSchema;
+		const result = await graphql({
+			schema, query, context, variables, validateGraphql: this.validateGraphql,
+		});
+
+		if (_.isEmpty(result.errors)) return result.data;
+
+		let fields = {};
+		const errors = result.errors;
+
+		errors.forEach((error) => {
+			error = this.formatError(error);
+			Object.assign(fields, error.fields);
+		});
+
+		// no user errors sent by server
+		if (!Object.keys(fields).length) {
+			fields = {
+				global: {
+					message: 'Unknown Error',
+					keyword: 'unknown',
+				},
+			};
+		}
+
+		const err = new GraphqlError(`[schema:${schemaName || 'default'}] Error in graphQL api`);
+		err.errors = errors;
+		err.fields = fields;
+		throw err;
+	}
+
+	async exec(query, context, {
```

Comment:
context could also be made optional since api doesn't require it, but don't provide defaults for it